### PR TITLE
Blocks: Canvas stories

### PR DESCRIPTION
**Pull request by JReinhold**
This PR adds stories for the Canvas block, both for current features and deprecated features.

Most notably, Having non-story elements/components within a Canvas as well as having multiple stories is being soft-deprecated. This means that we won't be documenting it officially (so in this PR the stories for those use cases are internal, and not in the public Blocks SB), but because we haven't quite figured out the migration path off of this yet* we aren't showing any deprecation warnings yet. This was discussed with @shilman, but I'm open to any input.

\* the migration path is likely the variants feature, which won't land for a while.

### CODE REVIEW

File: code/ui/blocks/src/blocks/Canvas.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+  },
+};
+export const WithSourceClosed: Story = {
+  args: {
+    withSource: 'closed',
+  },
+};
+
+// TODO: what is the purpose of mdxSource exactly?
+export const WithMdxSource: Story = {
```
Comment (JReinhold) [inline code/ui/blocks/src/blocks/Canvas.stories.tsx:38]: The `mdxSource` prop should be deprecated and renamed to something that falls in line with the naming in parameters, so `source` or `code`. I haven't quite figured out what the difference between the two yet.

any thoughts @tmeasday @shilman ?

File: code/ui/blocks/src/blocks/Canvas.tsx [hunk 2]
```diff
@@ -1,76 +1,66 @@
...
 export { SourceState };
 
-type CanvasProps = PurePreviewProps & {
+type CanvasProps = Omit<PurePreviewProps, 'isExpanded'> & {
   withSource?: SourceState;
   mdxSource?: string;
 };
 
-const getPreviewProps = (
+const usePreviewProps = (
```
Comment (JReinhold) [inline code/ui/blocks/src/blocks/Canvas.tsx:20]: It turned out this was actually a hook, just not named as such (which surfaced errors)

Comment (JReinhold) [inline code/ui/blocks/src/blocks/Canvas.tsx:20]: The diff for this function may seem like a lot, but it maintains the exact same functionality and logic, it's just a refactoring. The refactoring was sparked by the need for `useSourceProps` hook to be called unconditionally, and so in an attempt to figure the whole thing out I simplified the logic paths.

File: code/ui/blocks/src/blocks/Source.tsx [hunk 3]
```diff
@@ -89,7 +89,7 @@ const getSnippet = (snippet: string, story?: Store_Story<any>): string => {
 type SourceStateProps = { state: SourceState };
 type PureSourceProps = ComponentProps<typeof PureSource>;
 
-export const getSourceProps = (
+export const useSourceProps = (
```
Comment (JReinhold) [inline code/ui/blocks/src/blocks/Source.tsx:92]: also a hook, actually.

**Review by tmeasday — APPROVED** [review]
This is great, love the tests.

**Pull request merged**